mvc/views/tag-helpers/intro.rst

```diff
@@ -1,8 +1,281 @@
+
+ .. code-block:: html
+
+	<label class="caption" asp-for="FirstName"></label>
+ 
+ generates:
+ 
+ .. code-block:: html
+ 
+	<label class="caption" for="FirstName">First Name</label>
+
+The  camel-cased to sentence-cased content is not used if you add content to the ``<label>``. For example:
+
+ .. code-block:: html
+
+	<label class="caption" asp-for="FirstName">Name First</label>
+ 
+ generates:
+ 
+ .. code-block:: html
+ 
+	<label class="caption" for="FirstName">Name First</label>
+
+The following code image shows the Form portion of the *Views/Account/Register.cshtml* Razor view generated from the legacy ASP.NET 4.5.x MVC template included with Visual Studio 2015.
+
+.. image:: intro/_static/regCS.png 
+
+The Visual Studio editor displays C# code with a grey background. For example, the ``AntiForgeryToken`` HTML Helper:
+
+.. code-block:: html
+
+	@Html.AntiForgeryToken()
+ 
+is displayed with a grey background. Most of the markup in the Register view is C#. Compare that to the equivalent approach using Tag Helpers:
+ 
+.. image:: intro/_static/regTH.png 
+
+The markup is much cleaner and easier to read, edit, and maintain than the HTML Helpers approach. The C# code is reduced to the minimum that the server needs to know about. The Visual Studio editor displays markup targeted by a Tag Helper in a distinctive font. 
+
+Consider the *Email* group:
+
+.. literalinclude:: intro/sample/IntroToTagHelpers/src/IntroToTagHelpers/Views/Account/Register.cshtml
+   :language: c#
+   :lines: 12-18
+   :dedent: 4
+
+Each of the "asp-" attributes has a value of "Email", but "Email" is not a string. In this context, "Email" is the C# model expression property for the ``RegisterViewModel``. 
+
+
+The Visual Studio editor helps you write **all** of the markup in the Tag Helper approach of the register form, while Visual Studio provides no help for most of the code in the HTML Helpers approach. `IntelliSense support for Tag Helpers`_ goes into detail on working with Tag Helpers in the Visual Studio editor.
+
+Tag Helpers compared to Web Server Controls
+-----------------------------------------------
+
+Tag Helpers don't own the element they're associated with; they simply participate in the rendering of the element and content. ASP.NET `Web Server controls <https://msdn.microsoft.com/en-us/library/7698y1f0.aspx>`__ are declared and invoked on a page.
+
+`Web Server controls <https://msdn.microsoft.com/en-us/library/zsyt68f1.aspx>`__ have a non-trivial lifecycle that can make developing and debugging difficult. Web Server controls allow you to add functionality to the client Document Object Model (DOM) elements by using a client control. Tag Helpers have no DOM. Web Server controls include automatic browser detection and some have the ability to pass events from a nested control (such as a button in a table) to the container control.
```

Comment:
Should make this flow or break it into bulleted lists. As it reads right now it's more like an unbulleted bullet list.